### ISSUE TITLE
Failed to deploy Marblerun on Kubernetes cluster on an Azure Confidential Computing Node

### ISSUE THREAD

**Issue by ratnadeepb** [edgelesssys/marblerun#199]
### Issue description
Failed to deploy Marblerun on Kubernetes cluster on Azure SGX node. The coordinator fails to deploy with the error `Warning  FailedScheduling  30s (x9 over 7m34s)  default-scheduler  0/1 nodes are available: 1 Insufficient sgx.intel.com/epc.`

### To reproduce
```
curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add
sudo apt-add-repository "deb http://apt.kubernetes.io/ kubernetes-xenial main"
sudo apt install docker.io kubectl kubeadm kubelet build-essential autoconf gawk bison wget python3 libcurl4-openssl-dev python3-protobuf libprotobuf-c-dev protobuf-c-compiler python3-pip -y

sudo kubeadm init --apiserver-advertise-address 10.0.0.4 --pod-network-cidr=10.244.0.0/16
mkdir -p $HOME/.kube
sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
sudo chown $(id -u):$(id -g) $HOME/.kube/config

kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml

echo 'deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu bionic main' | sudo tee /etc/apt/sources.list.d/intel-sgx.list
wget -qO - https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | sudo apt-key add -
echo "deb [arch=amd64] https://packages.microsoft.com/ubuntu/18.04/prod bionic main" | sudo tee /etc/apt/sources.list.d/msprod.list
wget -qO - https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -

sudo apt update && sudo apt install -y libsgx-dcap-ql libsgx-dcap-ql-dev az-dcap-client

kubectl taint nodes --all node-role.kubernetes.io/master-

kubectl apply -k https://github.com/intel/intel-device-plugins-for-kubernetes/deployments/sgx_nfd?ref=main
kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.3.1/cert-manager.yaml
kubectl apply -k https://github.com/intel/intel-device-plugins-for-kubernetes/deployments/operator/default?ref=main

sudo wget -O /usr/local/bin/marblerun https://github.com/edgelesssys/marblerun/releases/latest/download/marblerun
sudo chmod +x /usr/local/bin/marblerun

sudo mkdir /mnt/data
kubectl apply -f storage_class.yaml
kubectl apply -f pv.yaml

marblerun install
```

### Expected behavior

Both the injector and the coordinator pods starting up. Only the injector pod starts up.

### Environment:
 - Marblerun version: 0.3.2
 - Edgeless RT version:
 - Go version:
 - Minikube version:
 - Kubernetes version: 1.21.2

**Comment by m1ghtym0** [comment]
Hi @ratnadeepb,
What does `kubectl describe node <node name> | grep sgx.intel.com` say?
Have you confirmed the [Intel demos](https://intel.github.io/intel-device-plugins-for-kubernetes/cmd/sgx_plugin/README.html#sgx-device-plugin-demos) work on your cluster?
Thanks!

**Comment by ratnadeepb** [comment]
I did not try the Intel demos. Anyhow, here is the result from the coordinator description:
```
Annotations:    sgx.intel.com/epc: 10
      sgx.intel.com/enclave:  1
      sgx.intel.com/epc:      10
      sgx.intel.com/enclave:  1
      sgx.intel.com/epc:      10
                             sgx.intel.com/epc:NoSchedule op=Exists
  Warning  FailedScheduling  60s (x2 over 62s)  default-scheduler  0/1 nodes are available: 1 Insufficient sgx.intel.com/enclave, 1 Insufficient sgx.intel.com/epc.
```

**Comment by m1ghtym0** [comment]
@ratnadeepb The coordinator is fine. But there is something wrong with your node and the sgx device plugin.
The command I posted gives information about the node. Additionally, information about the device plugin and its log would be great. Thanks!

**Comment by ratnadeepb** [comment]
I just realised I looked at the coordinator node instead of the node:
```
TestVM:~$ kubectl describe node testvm | grep sgx.intel.com
TestVM:~$ 
```

**Comment by daniel-weisse** [comment]
Hey @ratnadeepb
It seems your cluster has no sgx support in its current state.
Can you verify the VMs used in your cluster are of type Standard_DCXs_v2 ?

Microsoft also provides a tutorial for creating a cluster including device plugins / webhooks: https://docs.microsoft.com/en-us/azure/confidential-computing/confidential-nodes-aks-get-started
Following this tutorial should set up a cluster that is able to run Marblerun without problems

**Comment by m1ghtym0** [comment]
@ratnadeepb [Note that node feature discovery works only with the in-tree driver](https://intel.github.io/intel-device-plugins-for-kubernetes/cmd/sgx_plugin/README.html#deploy-node-feature-discovery).
Can you check which driver and kernel you're using?

**Comment by ratnadeepb** [comment]
```
TestVM:~$ uname -r
5.4.0-1051-azure
TestVM:~$ sudo lsmod | grep sgx
intel_sgx              32768  1
TestVM:~$ dmesg | grep sgx
[    6.834949] sgx: EPC section 0x2c0200000-0x2c37fffff
[    6.835360] sgx: EPC section 0x2c3a00000-0x2c3bfffff
[    6.835560] sgx: intel_sgx: Intel SGX DCAP Driver v1.33
```

**Comment by m1ghtym0** [comment]
I haven't found where they use the kernel config yet. But it might be worth giving this a try:

> Note: The default nfd-worker.conf assumes that the in-tree driver is used and enabled (CONFIG_X86_SGX=y). If the SGX DCAP out-of-tree driver is used, the kConfig match in nfd-worker.conf must be removed.

The log of the sgx device plugin would be interesting as well.

**Comment by ratnadeepb** [comment]
I redeployed the cluster after changing the `nfd-worker.conf` locally. Did not help. I am working on getting the SGX plugin logs.

Thanks for all the help.

**Comment by ratnadeepb** [comment]
Logs from the SGX device plugin for kubernetes:
`kube-rbac-proxy`:
```
TestVM:~$ kubectl logs -n inteldeviceplugins-system inteldeviceplugins-controller-manager-68bbdf7b9d-46fv9 kube-rbac-proxy
I0629 11:15:24.104702       1 main.go:186] Valid token audiences: 
I0629 11:15:24.104747       1 main.go:232] Generating self signed cert as no cert is provided
I0629 11:15:24.323771       1 main.go:281] Starting TCP socket on 0.0.0.0:8443
I0629 11:15:24.324551       1 main.go:288] Listening securely on 0.0.0.0:8443
```

`manager`:
```
TestVM:~$ kubectl logs -n inteldeviceplugins-system inteldeviceplugins-controller-manager-68bbdf7b9d-46fv9 manager
I0629 11:15:21.613145       1 request.go:668] Waited for 1.045855982s due to client-side throttling, not priority and fairness, request: GET:https://10.96.0.1:443/apis/deviceplugin.intel.com/v1?timeout=32s
I0629 11:15:21.617631       1 deleg.go:130] controller-runtime/metrics "msg"="metrics server is starting to listen"  "addr"="127.0.0.1:8080"
I0629 11:15:21.618377       1 deleg.go:130] controller-runtime/builder "msg"="Registering a mutating webhook"  "GVK"={"Group":"deviceplugin.intel.com","Version":"v1","Kind":"DsaDevicePlugin"} "path"="/mutate-deviceplugin-intel-com-v1-dsadeviceplugin"
I0629 11:15:21.618420       1 server.go:136] controller-runtime/webhook "msg"="registering webhook" "path"="/mutate-deviceplugin-intel-com-v1-dsadeviceplugin" 
I0629 11:15:21.618448       1 deleg.go:130] controller-runtime/builder "msg"="Registering a validating webhook"  "GVK"={"Group":"deviceplugin.intel.com","Version":"v1","Kind":"DsaDevicePlugin"} "path"="/validate-deviceplugin-intel-com-v1-dsadeviceplugin"
I0629 11:15:21.618467       1 server.go:136] controller-runtime/webhook "msg"="registering webhook" "path"="/validate-deviceplugin-intel-com-v1-dsadeviceplugin" 
I0629 11:15:21.618510       1 deleg.go:130] controller-runtime/builder "msg"="Registering a mutating webhook"  "GVK"={"Group":"deviceplugin.intel.com","Version":"v1","Kind":"FpgaDevicePlugin"} "path"="/mutate-deviceplugin-intel-com-v1-fpgadeviceplugin"
I0629 11:15:21.618528       1 server.go:136] controller-runtime/webhook "msg"="registering webhook" "path"="/mutate-deviceplugin-intel-com-v1-fpgadeviceplugin" 
I0629 11:15:21.618549       1 deleg.go:130] controller-runtime/builder "msg"="Registering a validating webhook"  "GVK"={"Group":"deviceplugin.intel.com","Version":"v1","Kind":"FpgaDevicePlugin"} "path"="/validate-deviceplugin-intel-com-v1-fpgadeviceplugin"
I0629 11:15:21.618566       1 server.go:136] controller-runtime/webhook "msg"="registering webhook" "path"="/validate-deviceplugin-intel-com-v1-fpgadeviceplugin" 
I0629 11:15:21.618606       1 deleg.go:130] controller-runtime/builder "msg"="Registering a mutating webhook"  "GVK"={"Group":"deviceplugin.intel.com","Version":"v1","Kind":"GpuDevicePlugin"} "path"="/mutate-deviceplugin-intel-com-v1-gpudeviceplugin"
I0629 11:15:21.618624       1 server.go:136] controller-runtime/webhook "msg"="registering webhook" "path"="/mutate-deviceplugin-intel-com-v1-gpudeviceplugin" 
I0629 11:15:21.618641       1 deleg.go:130] controller-runtime/builder "msg"="Registering a validating webhook"  "GVK"={"Group":"deviceplugin.intel.com","Version":"v1","Kind":"GpuDevicePlugin"} "path"="/validate-deviceplugin-intel-com-v1-gpudeviceplugin"
I0629 11:15:21.618657       1 server.go:136] controller-runtime/webhook "msg"="registering webhook" "path"="/validate-deviceplugin-intel-com-v1-gpudeviceplugin" 
I0629 11:15:21.618698       1 deleg.go:130] controller-runtime/builder "msg"="Registering a mutating webhook"  "GVK"={"Group":"deviceplugin.intel.com","Version":"v1","Kind":"QatDevicePlugin"} "path"="/mutate-deviceplugin-intel-com-v1-qatdeviceplugin"
I0629 11:15:21.618717       1 server.go:136] controller-runtime/webhook "msg"="registering webhook" "path"="/mutate-deviceplugin-intel-com-v1-qatdeviceplugin" 
I0629 11:15:21.618733       1 deleg.go:130] controller-runtime/builder "msg"="Registering a validating webhook"  "GVK"={"Group":"deviceplugin.intel.com","Version":"v1","Kind":"QatDevicePlugin"} "path"="/validate-deviceplugin-intel-com-v1-qatdeviceplugin"
I0629 11:15:21.618749       1 server.go:136] controller-runtime/webhook "msg"="registering webhook" "path"="/validate-deviceplugin-intel-com-v1-qatdeviceplugin" 
I0629 11:15:21.618786       1 deleg.go:130] controller-runtime/builder "msg"="Registering a mutating webhook"  "GVK"={"Group":"deviceplugin.intel.com","Version":"v1","Kind":"SgxDevicePlugin"} "path"="/mutate-deviceplugin-intel-com-v1-sgxdeviceplugin"
I0629 11:15:21.618807       1 server.go:136] controller-runtime/webhook "msg"="registering webhook" "path"="/mutate-deviceplugin-intel-com-v1-sgxdeviceplugin" 
I0629 11:15:21.618829       1 deleg.go:130] controller-runtime/builder "msg"="Registering a validating webhook"  "GVK"={"Group":"deviceplugin.intel.com","Version":"v1","Kind":"SgxDevicePlugin"} "path"="/validate-deviceplugin-intel-com-v1-sgxdeviceplugin"
I0629 11:15:21.618845       1 server.go:136] controller-runtime/webhook "msg"="registering webhook" "path"="/validate-deviceplugin-intel-com-v1-sgxdeviceplugin" 
I0629 11:15:21.618880       1 server.go:136] controller-runtime/webhook "msg"="registering webhook" "path"="/pods-sgx" 
I0629 11:15:21.618909       1 server.go:136] controller-runtime/webhook "msg"="registering webhook" "path"="/pods" 
I0629 11:15:21.618934       1 deleg.go:130] setup "msg"="starting manager"  
I0629 11:15:21.619016       1 leaderelection.go:243] attempting to acquire leader lease inteldeviceplugins-system/d1c7b6d5.intel.com...
I0629 11:15:21.619148       1 internal.go:384] intel-device-plugins-manager "msg"="starting metrics server"  "path"="/metrics"
I0629 11:15:21.619174       1 server.go:184] controller-runtime/webhook/webhooks "msg"="starting webhook server"  
I0629 11:15:21.619290       1 deleg.go:130] controller-runtime/certwatcher "msg"="Updated current TLS certificate"  
I0629 11:15:21.619383       1 deleg.go:130] controller-runtime/webhook "msg"="serving webhook server"  "host"="" "port"=9443
I0629 11:15:21.621532       1 deleg.go:130] controller-runtime/certwatcher "msg"="Starting certificate watcher"  
I0629 11:15:21.644775       1 leaderelection.go:253] successfully acquired lease inteldeviceplugins-system/d1c7b6d5.intel.com
I0629 11:15:21.720110       1 controller.go:165] intel-device-plugins-manager/controller/fpgadeviceplugin "msg"="Starting EventSource" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="FpgaDevicePlugin" "source"={"Type":{"metadata":{"creationTimestamp":null},"spec":{},"status":{"controlledDaemonSet":{},"desiredNumberScheduled":0,"numberReady":0}}}
I0629 11:15:21.720219       1 controller.go:165] intel-device-plugins-manager/controller/gpudeviceplugin "msg"="Starting EventSource" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="GpuDevicePlugin" "source"={"Type":{"metadata":{"creationTimestamp":null},"spec":{},"status":{"controlledDaemonSet":{},"desiredNumberScheduled":0,"numberReady":0}}}
I0629 11:15:21.720351       1 controller.go:165] intel-device-plugins-manager/controller/sgxdeviceplugin "msg"="Starting EventSource" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="SgxDevicePlugin" "source"={"Type":{"metadata":{"creationTimestamp":null},"spec":{},"status":{"controlledDaemonSet":{},"desiredNumberScheduled":0,"numberReady":0}}}
I0629 11:15:21.720373       1 controller.go:165] intel-device-plugins-manager/controller/acceleratorfunction "msg"="Starting EventSource" "reconciler group"="fpga.intel.com" "reconciler kind"="AcceleratorFunction" "source"={"Type":{"metadata":{"creationTimestamp":null},"status":{},"spec":{"afuId":"","interfaceId":"","mode":""}}}
I0629 11:15:21.720384       1 controller.go:173] intel-device-plugins-manager/controller/acceleratorfunction "msg"="Starting Controller" "reconciler group"="fpga.intel.com" "reconciler kind"="AcceleratorFunction" 
I0629 11:15:21.720434       1 controller.go:165] intel-device-plugins-manager/controller/dsadeviceplugin "msg"="Starting EventSource" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="DsaDevicePlugin" "source"={"Type":{"metadata":{"creationTimestamp":null},"spec":{},"status":{"controlledDaemonSet":{},"desiredNumberScheduled":0,"numberReady":0}}}
I0629 11:15:21.720525       1 controller.go:165] intel-device-plugins-manager/controller/qatdeviceplugin "msg"="Starting EventSource" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="QatDevicePlugin" "source"={"Type":{"metadata":{"creationTimestamp":null},"spec":{},"status":{"controlledDaemonSet":{},"desiredNumberScheduled":0,"numberReady":0}}}
I0629 11:15:21.720631       1 controller.go:165] intel-device-plugins-manager/controller/fpgaregion "msg"="Starting EventSource" "reconciler group"="fpga.intel.com" "reconciler kind"="FpgaRegion" "source"={"Type":{"metadata":{"creationTimestamp":null},"status":{},"spec":{"interfaceId":""}}}
I0629 11:15:21.720645       1 controller.go:173] intel-device-plugins-manager/controller/fpgaregion "msg"="Starting Controller" "reconciler group"="fpga.intel.com" "reconciler kind"="FpgaRegion" 
I0629 11:15:21.721283       1 controller.go:165] intel-device-plugins-manager/controller/fpgadeviceplugin "msg"="Starting EventSource" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="FpgaDevicePlugin" "source"={"Type":{"metadata":{"creationTimestamp":null},"spec":{"selector":null,"template":{"metadata":{"creationTimestamp":null},"spec":{"containers":null}},"updateStrategy":{}},"status":{"currentNumberScheduled":0,"numberMisscheduled":0,"desiredNumberScheduled":0,"numberReady":0}}}
I0629 11:15:21.721365       1 controller.go:173] intel-device-plugins-manager/controller/fpgadeviceplugin "msg"="Starting Controller" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="FpgaDevicePlugin" 
I0629 11:15:21.721707       1 controller.go:165] intel-device-plugins-manager/controller/gpudeviceplugin "msg"="Starting EventSource" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="GpuDevicePlugin" "source"={"Type":{"metadata":{"creationTimestamp":null},"spec":{"selector":null,"template":{"metadata":{"creationTimestamp":null},"spec":{"containers":null}},"updateStrategy":{}},"status":{"currentNumberScheduled":0,"numberMisscheduled":0,"desiredNumberScheduled":0,"numberReady":0}}}
I0629 11:15:21.721719       1 controller.go:173] intel-device-plugins-manager/controller/gpudeviceplugin "msg"="Starting Controller" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="GpuDevicePlugin" 
I0629 11:15:21.721734       1 controller.go:165] intel-device-plugins-manager/controller/sgxdeviceplugin "msg"="Starting EventSource" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="SgxDevicePlugin" "source"={"Type":{"metadata":{"creationTimestamp":null},"spec":{"selector":null,"template":{"metadata":{"creationTimestamp":null},"spec":{"containers":null}},"updateStrategy":{}},"status":{"currentNumberScheduled":0,"numberMisscheduled":0,"desiredNumberScheduled":0,"numberReady":0}}}
I0629 11:15:21.721742       1 controller.go:173] intel-device-plugins-manager/controller/sgxdeviceplugin "msg"="Starting Controller" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="SgxDevicePlugin" 
I0629 11:15:21.721775       1 controller.go:165] intel-device-plugins-manager/controller/dsadeviceplugin "msg"="Starting EventSource" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="DsaDevicePlugin" "source"={"Type":{"metadata":{"creationTimestamp":null},"spec":{"selector":null,"template":{"metadata":{"creationTimestamp":null},"spec":{"containers":null}},"updateStrategy":{}},"status":{"currentNumberScheduled":0,"numberMisscheduled":0,"desiredNumberScheduled":0,"numberReady":0}}}
I0629 11:15:21.721797       1 controller.go:173] intel-device-plugins-manager/controller/dsadeviceplugin "msg"="Starting Controller" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="DsaDevicePlugin" 
I0629 11:15:21.721806       1 controller.go:165] intel-device-plugins-manager/controller/qatdeviceplugin "msg"="Starting EventSource" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="QatDevicePlugin" "source"={"Type":{"metadata":{"creationTimestamp":null},"spec":{"selector":null,"template":{"metadata":{"creationTimestamp":null},"spec":{"containers":null}},"updateStrategy":{}},"status":{"currentNumberScheduled":0,"numberMisscheduled":0,"desiredNumberScheduled":0,"numberReady":0}}}
I0629 11:15:21.721812       1 controller.go:173] intel-device-plugins-manager/controller/qatdeviceplugin "msg"="Starting Controller" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="QatDevicePlugin" 
I0629 11:15:21.822189       1 controller.go:207] intel-device-plugins-manager/controller/qatdeviceplugin "msg"="Starting workers" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="QatDevicePlugin" "worker count"=1
I0629 11:15:21.822189       1 controller.go:207] intel-device-plugins-manager/controller/sgxdeviceplugin "msg"="Starting workers" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="SgxDevicePlugin" "worker count"=1
I0629 11:15:21.822210       1 controller.go:207] intel-device-plugins-manager/controller/fpgadeviceplugin "msg"="Starting workers" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="FpgaDevicePlugin" "worker count"=1
I0629 11:15:21.822218       1 controller.go:207] intel-device-plugins-manager/controller/gpudeviceplugin "msg"="Starting workers" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="GpuDevicePlugin" "worker count"=1
I0629 11:15:21.822242       1 controller.go:207] intel-device-plugins-manager/controller/dsadeviceplugin "msg"="Starting workers" "reconciler group"="deviceplugin.intel.com" "reconciler kind"="DsaDevicePlugin" "worker count"=1
I0629 11:15:21.823294       1 controller.go:207] intel-device-plugins-manager/controller/acceleratorfunction "msg"="Starting workers" "reconciler group"="fpga.intel.com" "reconciler kind"="AcceleratorFunction" "worker count"=1
I0629 11:15:21.823297       1 controller.go:207] intel-device-plugins-manager/controller/fpgaregion "msg"="Starting workers" "reconciler group"="fpga.intel.com" "reconciler kind"="FpgaRegion" "worker count"=1
```

**Comment by ratnadeepb** [comment]
I missed this the last time. Attempting to deploy the daemonset - `TestVM:~$ kubectl apply -k deployments/sgx_plugin/overlays/epc-nfd/`:
```
Type     Reason       Age               From               Message
  ----     ------       ----              ----               -------
  Normal   Scheduled    34s               default-scheduler  Successfully assigned default/intel-sgx-plugin-xcnkf to testvm
  Warning  FailedMount  3s (x7 over 35s)  kubelet            MountVolume.SetUp failed for volume "sgx-enclave" : hostPath type check failed: /dev/sgx_enclave is not a character device
  Warning  FailedMount  3s (x7 over 35s)  kubelet            MountVolume.SetUp failed for volume "sgx-provision" : hostPath type check failed: /dev/sgx_provision is not a character device
```

```
estVM:~$ ls /dev/sgx
enclave  provision
```

**Comment by ratnadeepb** [comment]
The udev [rules ](https://intel.github.io/intel-device-plugins-for-kubernetes/cmd/sgx_plugin/README.html#backwards-compatibility-note)exist but are not taking effect - not sure why though!

**Comment by ratnadeepb** [comment]
I think I resolved the issue with:
```
sudo ln -s /dev/sgx/enclave /dev/sgx_enclave
sudo ln -s /dev/sgx/provision /dev/sgx_provision
```

`kubectl describe node testvm | grep sgx`:
```
TestVM:~$ kubectl describe node testvm | grep sgx
                    nfd.node.kubernetes.io/extended-resources: sgx.intel.com/epc
  sgx.intel.com/enclave:    110
  sgx.intel.com/epc:        58720256
  sgx.intel.com/provision:  110
  sgx.intel.com/enclave:    110
  sgx.intel.com/epc:        58720256
  sgx.intel.com/provision:  110
  default                     intel-sgx-plugin-mj6gl                                    0 (0%)        0 (0%)      0 (0%)           0 (0%)         3m30s
  sgx.intel.com/enclave    1            1
  sgx.intel.com/epc        10           10
  sgx.intel.com/provision  0            0
```

`TestVM:~$ kubectl describe pods -n marblerun marblerun-coordinator-d4cdb57cd-8sw76`:
```
Events:
  Type     Reason     Age   From               Message
  ----     ------     ----  ----               -------
  Normal   Scheduled  3m6s  default-scheduler  Successfully assigned marblerun/marblerun-coordinator-d4cdb57cd-8sw76 to testvm
  Normal   Pulled     3m5s  kubelet            Container image "ghcr.io/edgelesssys/coordinator:v0.3.2" already present on machine
  Normal   Created    3m4s  kubelet            Created container coordinator
  Normal   Started    3m4s  kubelet            Started container coordinator
  Warning  Unhealthy  3m4s  kubelet            Readiness probe failed: Get "https://10.244.0.22:4433/status": dial tcp 10.244.0.22:4433: connect: connection refused
```

**Comment by m1ghtym0** [comment]
Ah, I didn't read your output carefully enough.
You're using the sgx driver version 1.33 (probably preinstalled in your Azure VM).
The Intel device plugin requires the in-tree kernel driver from 5.11 or at least 1.41:

> The SGX device plugin requires Linux Kernel SGX drivers to be available. These drivers are available in Linux 5.11. The SGX DCAP out-of-tree driver v1.41 is also known to work.

They provide [backward compatibility](https://intel.github.io/intel-device-plugins-for-kubernetes/cmd/sgx_plugin/README.html#backwards-compatibility-note) the way you solved it now.

Can I close this issue?

**Comment by ratnadeepb** [comment]
Thanks for all the help
